xgi/classes/reportviews.py

```diff
@@ -149,18 +168,14 @@ def __getitem__(self, id):
             the node does not exist in the hypergraph.
 
         """
+        if id not in self._ids:
+            raise XGIError(f"The ID {id} is not in this view")
+
         try:
-            return self._id_attrs[id]
+            return self._id_attr[id]
         except KeyError as e:
             raise XGIError(f"The ID {id} is not in the hypergraph") from e
```

Comment:
I would also add a TypeError for when a user specifies an unhashable id

Reply:
done!